resources/js/app/components/permission/permission.vue

```diff
@@ -0,0 +1,41 @@
+<template>
+    <span v-if="hasPermission()" v-title="title()">
+        <Icon icon="carbon:locked" v-if="isLocked()"></Icon>
+        <Icon icon="carbon:unlocked" v-if="!isLocked()"></Icon>
+        <Icon icon="carbon:tree-view" v-if="inherited"></Icon>
+    </span>
+</template>
+
+<script>
+import { t } from 'ttag';
+
+export default {
+
+    name: 'permission',
+
+    props: {
+        inherited: false,
+        objectRoles: [],
+        role: '',
+        userRoles: [],
+    },
+
+    data() {
+        return {
+            msgInherited: t`Inherited`,
+        };
+    },
+
+    methods: {
```

Comment:
```suggestion
    computed: {
```
These are good candidates for computed properties, instead of being methods. The difference is that the result of the function is cached based on the property used (ex. result of `isLocked` is updated only when `this.userRoles` or `this.role` change). If you change this, you can use them in template as normal properties (ex. `isLocked()` becomes `isLocked`)